src/combinator/limit.js

```diff
@@ -110,6 +110,7 @@ DebounceSink.prototype.error = function (t, x) {
 
 DebounceSink.prototype.dispose = function () {
   this._clearTimer()
+  this.disposable.dispose()
```

Comment:
Yikes.  Good catch, @ppoliani.  This just needs a return, `return this.disposable.dispose()` since disposables typically form a promise chain in the current version of most.js.